From 42e0694555f80ed93ff2013580d805fa1d928ff7 Mon Sep 17 00:00:00 2001
From: Steve Lhomme <robux4@ycbcr.xyz>
Date: Mon, 16 Dec 2024 10:01:27 +0100
Subject: [PATCH 3/5] win32: define struct timezone for non-mingw-w32

The compiler complains if the structure used in the .c file is defined after an
anonymous version was used in the header.
---
 include/win32/win32_compat.h | 8 +++++++-
 win32/win32_compat.c         | 6 ------
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/include/win32/win32_compat.h b/include/win32/win32_compat.h
index 70214b7..e11bf70 100755
--- a/include/win32/win32_compat.h
+++ b/include/win32/win32_compat.h
@@ -150,9 +150,15 @@ struct pollfd {
 int     getpid(void);
 int     win32_inet_pton(int af, const char * src, void * dst);
 int     win32_poll(struct pollfd *fds, unsigned int nfsd, int timeout);
-int     win32_gettimeofday(struct timeval *tv, struct timezone *tz);
 #ifdef __MINGW32__
 # define win32_gettimeofday mingw_gettimeofday
+#else
+struct timezone
+{
+  int  tz_minuteswest; /* minutes W of Greenwich */
+  int  tz_dsttime;     /* type of dst correction */
+};
+int     win32_gettimeofday(struct timeval *tv, struct timezone *tz);
 #endif
 
 #define DllExport
diff --git a/win32/win32_compat.c b/win32/win32_compat.c
index 3654788..fb5f0aa 100644
--- a/win32/win32_compat.c
+++ b/win32/win32_compat.c
@@ -157,12 +157,6 @@ int win32_poll(struct pollfd *fds, unsigned int nfds, int timo)
 #endif
  
 #ifndef __MINGW32__
-struct timezone 
-{
-  int  tz_minuteswest; /* minutes W of Greenwich */
-  int  tz_dsttime;     /* type of dst correction */
-};
- 
 int win32_gettimeofday(struct timeval *tv, struct timezone *tz)
 {
   FILETIME ft;
-- 
2.45.0.windows.1

